src/sentry/utils/kafka_config.py

```diff
@@ -0,0 +1,72 @@
+from __future__ import absolute_import
+import six
+from django.conf import settings
+
+SUPPORTED_KAFKA_CONFIGURATION = (
+    # Check https://github.com/edenhill/librdkafka/blob/master/CONFIGURATION.md
+    # for the full list of available options
+    "bootstrap.servers",
+    "compression.type",
+    "message.max.bytes",
+    "sasl.mechanism",
+    "sasl.username",
+    "sasl.password",
+    "security.protocol",
+)
+COMMON_SECTION = "common"
+PRODUCERS_SECTION = "producers"
+CONSUMERS_SECTION = "consumers"
+ADMIN_SECTION = "admin"
+KNOWN_SECTIONS = (COMMON_SECTION, PRODUCERS_SECTION, CONSUMERS_SECTION, ADMIN_SECTION)
+
+
+def _get_legacy_kafka_cluster_options(cluster_name):
+    options = settings.KAFKA_CLUSTERS[cluster_name]
+
+    options = {k: v for k, v in options.items() if k not in KNOWN_SECTIONS}
+    if "bootstrap.servers" in options:
+        if isinstance(options["bootstrap.servers"], (list, tuple)):
+            options["bootstrap.servers"] = ",".join(options["bootstrap.servers"])
+    return options
+
+
+def _get_kafka_cluster_options(
+    cluster_name, config_section, with_legacy=False, override_params=None
+):
+    options = {}
+    custom_options = settings.KAFKA_CLUSTERS[cluster_name].get(config_section, {})
+    common_options = settings.KAFKA_CLUSTERS[cluster_name].get(COMMON_SECTION, {})
+    legacy_options = _get_legacy_kafka_cluster_options(cluster_name)
+    if with_legacy and legacy_options:
+        # we prefer these ones
+        options.update(legacy_options)
+    else:
+        options.update(common_options)
+        options.update(custom_options)
+    # check key validity
+    for configuration_key in options:
+        if configuration_key not in SUPPORTED_KAFKA_CONFIGURATION:
+            raise ValueError(
+                "The `{configuration_key}` configuration key is not supported.".format(
+                    configuration_key=configuration_key
+                )
+            )
+    if not isinstance(options["bootstrap.servers"], six.string_types):
+        raise ValueError("bootstrap.servers must be a comma separated string")
```

Comment:
Up to you,
I would not block the PR for this: I think it would me more intuitive (for the new configuration, where this is possible) to have the bootstrap servers in the config represented as a sequence (harder to mess up the structure), and then turn it into a string (here in this module) instead of doing the opposite. This would follow the same convention we have in snuba.

Unfortunately we still need to support the strings for the legacy config, but that, being legacy will be eventually removed, so we can eventually get to a consistent representation.

Reply:
at this stage in code, the bootstrap.servers must be a comma separated string, as this is what librdkafka is expected. 
sure we can make it a list before (same as it was in the legacy config prior these changes)